proposals/infra/INF-0004-inproc-outofproc-compiler-library-support.md

```diff
@@ -0,0 +1,113 @@
+<!-- {% raw %} -->
+
+# Support for in-process, Out-of-process, or both for the c style compiler api
+
+* Proposal: [NNNN](INF-NNNN-inproc-outofproc-compiler-api-support.md)
+* Author(s): [Cooper Partin](https://github.com/coopp)
+* Sponsor: [Cooper Partin](https://github.com/coopp)
+* Status: **Under Consideration**
+* Impacted Project(s): (Clang)
+
+* Issues:
+
+## Introduction
+
+An effort is underway to bring compilation support into clang for HLSL based
+shaders.  A C style api will be built to enable applications to compile
+a shader from their own processes in addition to being able to launching the
+clang process.
+
+This document is to propose building only the out-of-process support which
+gives the caller a more performant and secure solution.
+
+Visual Studio uses an out of process architecture for compilation which helps
+better utilize the power of the machine performing the compilation as well as
+isolating the code editor from any crashes that may occur during compilation.
+
+### Overview of api hosting differences
+Here are two common ways a library can be loaded and used.
+
+* **In Process** - An application calls an api that loads compilation support
+into the application's process. All work is performed within the application's
+process.
+
+* **Out of Process** - An application calls an api that launches one or more
+processes to perform the compilation. All work is performed outside the
+calling application's process.
+
+## Motivation
+
+### Why prefer an out of process solution?
```

Comment:
Also:
* cleanup between compiler invocations
* handling of catastrophic errors (if compiler aborts on error then what does that do to the calling process?)

Reply:
Fixed